eth2util/signing/signing.go

```diff
@@ -128,121 +126,111 @@ func GetDataRoot(ctx context.Context, eth2Cl Eth2Provider, name DomainName, epoc
 //
 //}
 
-// VerifySignedData verifies any eth2 signed data signature.
-// If it is partially signed, provide the pubshare.
-// If it is aggregate signed, provide the group pubkey.
-func VerifySignedData(ctx context.Context, eth2Cl Eth2Provider, pubkey *bls_sig.PublicKey,
-	eth2SignedType interface{},
-) error {
-	// eth2util shouldn't import core package, so can't use core.SignedData.
-	// To avoid pointer vs non-pointer issues, always get value if pointer is provided.
-	if reflect.TypeOf(eth2SignedType).Kind() == reflect.Pointer {
-		eth2SignedType = reflect.ValueOf(eth2SignedType).Elem().Interface()
+func VerifyAttestation(ctx context.Context, eth2Cl Eth2Provider, pubkey *bls_sig.PublicKey, att eth2p0.Attestation) error {
```

Comment:
I think all these eth2 types should be pointers, since that is how goeth2 users them, always as pointers.

Reply:
sure